src/frontend/dsl/mod.rs

```diff
@@ -82,6 +82,12 @@ impl<F, TG: TraceGenerator<F>> CircuitContext<F, TG> {
         Queriable::Halo2FixedQuery(self.circuit.add_halo2_fixed(name, column), 0)
     }
 
+    /// Imports a halo2 table column with a name string into the circuit and returns a
+    /// `Queriable` instance representing the imported column.
+    pub fn import_halo2_table(&mut self, name: &str, column: TableColumn) -> Queriable<F> {
+        Queriable::Halo2FixedQuery(self.circuit.add_halo2_table(name, column), 0)
```

Comment:
@alxkzmn there should be also a new type of query, Halo2TableQuery. Why not?

Reply:
Regarding this - I have been trying a different approach that is not using any Halo2 frontend code at all. Right now we need the TableColumns only because we are still using the CosntraintSystem from the Halo2 frontend. Later it is converted to CosntraintSystemMid which is in the middleware. It is possible to create the CosntraintSystemMid directly by mapping the chiquito poly expressions and lookups to halo2 equivalents (that is currently happening inside the frontend ConstraintSystem). I have consulted with Edu and he thinks that it's feasible without writing much new code (also there are some examples of how it's done from plonky3 frontend to halo2 middleware).

Reply:
So, this makes me confused. I thought we were eliminating the halo2 front-end entirely. I am unsure what we should do. If we are going to still keep using ContraintSystem then I think the approach I am describing is the best, to separate the columns between fixed and table all the way.

Reply:
So, in that case we only need to do the guessing for the native chiquito fixed columns